src/main/java/com/specialwarriors/conal/common/auth/jwt/JwtProvider.java

```diff
@@ -0,0 +1,81 @@
+package com.specialwarriors.conal.common.auth.jwt;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.ExpiredJwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.Jwts.SIG;
+import java.nio.charset.StandardCharsets;
+import java.util.Date;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class JwtProvider {
+
+    private final SecretKey secretKey;
+    @Value("${spring.jwt.access-token-exp}")
+    private long accessTokenExp;
+    @Value("${spring.jwt.refresh-token-exp}")
+    private long refreshTokenExp;
+
+    public JwtProvider(@Value("${spring.jwt.secret-key}") String secret) {
+        secretKey = new SecretKeySpec(secret.getBytes(StandardCharsets.UTF_8),
+                SIG.HS256.key().build().getAlgorithm());
+    }
+
+    public JwtTokenResponse generateTokens(Long userId) {
+        return new JwtTokenResponse(createAccessToken(userId), createRefreshToken(userId));
+    }
+
+    public String createAccessToken(Long userId) {
+        Date now = new Date();
+        Date accessExp = new Date(now.getTime() + accessTokenExp);
+
+        return Jwts.builder()
+                .subject(String.valueOf(userId))
+                .issuedAt(now)
+                .expiration(accessExp)
+                .signWith(secretKey)
+                .compact();
+    }
+
+    public String createRefreshToken(Long userId) {
+        Date now = new Date();
+        Date refreshExp = new Date(now.getTime() + refreshTokenExp);
+
+        return Jwts.builder()
+                .subject(String.valueOf(userId))
+                .issuedAt(now)
+                .expiration(refreshExp)
+                .signWith(secretKey)
+                .compact();
+    }
+
+    public boolean validateToken(String token) {
+        try {
+            Jwts.parser().verifyWith(secretKey).build().parseSignedClaims(token);
+            return true;
+        } catch (ExpiredJwtException e) {
+            return false;
+        }
+    }
+
+    public boolean isExpired(String token) {
+        try {
+            Claims claims = Jwts.parser().verifyWith(secretKey).build()
+                    .parseSignedClaims(token).getPayload();
+            return claims.getExpiration().before(new Date());
+        } catch (ExpiredJwtException e) {
+            return true;
+        }
+    }
```

Comment:
이하 사항들을 고려해보시길 부탁 드립니다.
- `parseSignedClaims()` 자체가 만료된 토큰일 경우 예외를 던짐 → 사실상 `getExpiration()`까지 가지 않음
- `ExpiredJwtException`이 발생한 시점에서 이미`true`를 반환하는 것이므로 `claims.getExpiration()`은 무의미

Reply:
토큰 재발행 로직에서 refreshToken을 db에 저장한 후 불러오기 때문에 유효성 검증은 불필요하다고 생각해 validate와 isExpired 메서드로 나눴습니다.
그러나 말씀주신대로 `parseSignedClaims()`에서 예외를 던지므로 try문 내부의 return문은 삭제하고, try-catch문이 끝난 후에 return false를 반환하도록 수정하겠습니다.
```suggestion
    public boolean isExpired(String token) {
        try {
            Jwts.parser().verifyWith(secretKey).build()
                    .parseSignedClaims(token).getPayload();
        } catch (ExpiredJwtException e) {
            return true;
        }
        return false;
    }
```